lib/srv/sess.go

```diff
@@ -211,6 +211,25 @@ func (s *SessionRegistry) OpenSession(ch ssh.Channel, req *ssh.Request, ctx *Ser
 
 		return nil
 	}
+
+	if ctx.srv.GetCreateHostUser() &&
+		ctx.Identity.RoleSet.CreateHostUser(s.srv.GetInfo()) {
+		mgmt, err := NewUserManagement()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		userCloser, err := createTemporaryUser(mgmt,
```

Comment:
I'd make whatever `mgmt` here into an interface and have methods like `Create`, `Delete`, etc. on it.